mr/src/main/java/org/elasticsearch/hadoop/serialization/bulk/AbstractBulkFactory.java

```diff
@@ -120,6 +121,12 @@ void doWrite(Object value) {
 
                 pool.get().bytes(valueString);
             }
+            
+            else if (value instanceof Date) {
+                String valueString = (value == null ? "null": Long.toString(((Date) value).getTime()));
+                pool.get().bytes(valueString);
+            }
```

Comment:
I'll suspend this critique then. We're already doing this same logic for Strings, Numbers, Booleans, and nulls. I'll open a separate issue to address this